terraform/main.tf

```diff
@@ -0,0 +1,156 @@
+/**
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Declares the variable "services" which lists the Google Cloud services to be enabled.
+variable "services" {
+  type    = list(string)
+  default = ["artifactregistry.googleapis.com", "cloudbuild.googleapis.com", "run.googleapis.com", "cloudscheduler.googleapis.com", "iam.googleapis.com"]
+}
+
+// Enables the Google Cloud services listed in the "services" variable for the project specified.
+resource "google_project_service" "project_service" {
```

Comment:
Small follow up question. I did a test deployment and, as expected, these APIs were disabled when running `terraform destroy` to tear everything down.

The thing is they get disabled even if they were already enabled and there are other resources in them. For example, I had a few Artifact Registry repositories for other uses and the API got disabled anyway.

Coming from a far from Terraform expert, is there anyway we can avoid that?